freqtrade/tests/test_backtesting.py

```diff
@@ -54,12 +102,43 @@ def backtest(backtest_conf, backdata, mocker):
     return results
 
 
-@pytest.mark.skipif(not os.environ.get('BACKTEST', False), reason="BACKTEST not set")
-def test_backtest(backtest_conf, backdata, mocker, report=True):
-    results = backtest(backtest_conf, backdata, mocker)
+@pytest.mark.skipif(not os.environ.get('BACKTEST'), reason="BACKTEST not set")
+def test_backtest(backtest_conf, mocker):
+    print('')
+
+    config = None
+    conf_path = os.environ.get('BACKTEST_CONFIG')
+    if conf_path:
+        print('Using config: {} ...'.format(conf_path))
+        with open(conf_path, 'r') as conf_file:
+            config = json.load(conf_file)
+
+    ticker_interval = int(os.environ.get('BACKTEST_TICKER_INTERVAL') or 5)
+    print('Using ticker_interval: {} ...'.format(ticker_interval))
+
+    data = {}
+    if os.environ.get('BACKTEST_LIVE'):
+        print('Downloading data for all pairs in whitelist ...')
+        exchange._API = Bittrex({'key': '', 'secret': ''})
+        for pair in config['exchange']['pair_whitelist']:
+            data[pair] = exchange.get_ticker_history(pair, ticker_interval)
+    else:
+        print('Using local backtesting data (ignoring whitelist in given config)...')
+        data = load_backtesting_data(ticker_interval)
+
+    config = config or backtest_conf
+
+    print('Using stake_currency: {} ...\nUsing stake_amount: {} ...'.format(
+        config['stake_currency'], config['stake_amount']
+    ))
+
+    min_date, max_date = get_timeframe(data)
+    print('Measuring data from {} up to {} ...'.format(
+        min_date.isoformat(), max_date.isoformat()
+    ))
```

Comment:
this function is growing too large.

Reply:
I created a shared function to load the config file, but I don't think we can reduce it more than that. The complexity is very low and half of the code contains logging